### PR TITLE
feat: [CM-607][Sale Widget] Get order config from product skus + qty

### PR DESCRIPTION
**Pull request by jhesgodi**
# Summary
Improve how primary sales widget gets the quote and configurations for an order. Instead of the widget knowing which `amount` needs to be paid, it can now instead pass the items `sku` & `qty` to the `/client-config` service, and it will be able to return a response with by item amounts and total amounts including conversions.

**BREAKING CHANGE:**
Requires customers to implement new [Quotes endpoint](https://docs.immutable.com/docs/zkEVM/products/checkout/widgets/primary-sales/backend/byo#quote-endpoint)

# Screenshots
**Before:**
Screenshot 2024-05-09 at 11 18 29 AM

**After: Multi item with correct pricing conversion**
Screenshot 2024-05-08 at 1 39 15 PM

# Detail and impact of the change
## Added 
- Able to list multiple products multiple currency
- Able to quote a price by having the items sku and qty

## Changed
- No longer required to be initialised with the `amount` param

## Removed
- `amount` param

## Fixed
- Item pricing is display correct in the settlement currency selected by customer

### CODE REVIEW

File: packages/checkout/widgets-lib/src/widgets/sale/hooks/useSmartCheckout.ts [hunk 1]
```diff
@@ -15,17 +15,17 @@ type UseSmartCheckoutInput = {
...
 
 export const useSmartCheckout = ({
   checkout,
   provider,
   items,
-  amount,
+  
   tokenAddress,
 }: UseSmartCheckoutInput) => {
+  const amount = '0';
```
Comment (jhesgodi) [inline packages/checkout/widgets-lib/src/widgets/sale/hooks/useSmartCheckout.ts:28]: Please ignore this patch
The entire hook gets remove in next PR as is not longer required for multi currency flow.

**Review by jwhardwick — DISMISSED** [review]
Looks good 🔥

**Review by luads — APPROVED** [review]

**Review by dreamoftrees — APPROVED** [review]
Great work @jhesgodi

**Review by mimi-imtbl — APPROVED** [review]

**Review by jwhardwick — DISMISSED** [review]

**Review by luads — APPROVED** [review]

**Review by luads — APPROVED** [review]

**Review by mimi-imtbl — APPROVED** [review]

**Review by dreamoftrees — APPROVED** [review]
👍

**Review by jwhardwick — APPROVED** [review]

**Pull request merged**